scripts/zkai_may2026_competitor_metric_matrix_gate.py

```
@@ -0,0 +1,654 @@
+        raise CompetitorMetricMatrixError(f"TSV source has duplicate columns: {duplicate_fields}")
+    fieldnames = set(fieldname_list)
+    missing = REQUIRED_PUBLISHED_COLUMNS - fieldnames
+    if missing:
+        raise CompetitorMetricMatrixError(f"TSV source missing columns: {sorted(missing)}")
+    extra = fieldnames - REQUIRED_PUBLISHED_COLUMNS
+    if extra:
+        raise CompetitorMetricMatrixError(f"TSV source has extra columns: {sorted(extra)}")
+    rows = list(reader)
+    for index, row in enumerate(rows, start=2):
+        if None in row:
+            raise CompetitorMetricMatrixError(f"TSV source row {index} has extra cells")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)